pq-crypto/sike_r3/sikep434r3_fp.c

```
@@ -0,0 +1,305 @@
+* If ma < 2^448*p434, the output mc is in the range [0, 2*p434-1].
+* ma is assumed to be in Montgomery representation. */
+void rdc_mont(digit_t* ma, digit_t* mc)
+{
+  /* Assembly code take out delete later
+#if defined(SIKE_P434_R3_ASM)
+  if (sikep434r3_asm_is_enabled()) {
+      rdc434_asm(ma, mc);
+      return;
+  }
+#endif*/
+
+  unsigned int i, j, carry, count = SIKE_P434_R3_ZERO_WORDS;
+  digit_t UV[2], t = 0, u = 0, v = 0;
+
+  for (i = 0; i < SIKE_P434_R3_NWORDS_FIELD; i++) {
+      mc[i] = 0;
+  }
+
+  for (i = 0; i < SIKE_P434_R3_NWORDS_FIELD; i++) {
+      for (j = 0; j < i; j++) {
+          if (j < (i-SIKE_P434_R3_ZERO_WORDS+1)) {
+              SIKE_P434_R3_MUL(mc[j], ((const digit_t*)p434p1)[i-j], UV+1, UV[0]);
+              SIKE_P434_R3_ADDC(0, UV[0], v, carry, v);
+              SIKE_P434_R3_ADDC(carry, UV[1], u, carry, u);
+              t += carry;
+          }
+      }
+      SIKE_P434_R3_ADDC(0, v, ma[i], carry, v);
+      SIKE_P434_R3_ADDC(carry, u, 0, carry, u);
+      t += carry;
+      mc[i] = v;
+      v = u;
+      u = t;
+      t = 0;
+  }
+
+  for (i = SIKE_P434_R3_NWORDS_FIELD; i < 2*SIKE_P434_R3_NWORDS_FIELD-1; i++) {
+      if (count > 0) {
+          count -= 1;
+      }
+      for (j = i-SIKE_P434_R3_NWORDS_FIELD+1; j < SIKE_P434_R3_NWORDS_FIELD; j++) {
+          if (j < (SIKE_P434_R3_NWORDS_FIELD-count)) {
+              SIKE_P434_R3_MUL(mc[j], ((const digit_t*)p434p1)[i-j], UV+1, UV[0]);
+              SIKE_P434_R3_ADDC(0, UV[0], v, carry, v);
+              SIKE_P434_R3_ADDC(carry, UV[1], u, carry, u);
+              t += carry;
+          }
+      }
+      SIKE_P434_R3_ADDC(0, v, ma[i], carry, v);
+      SIKE_P434_R3_ADDC(carry, u, 0, carry, u);
+      t += carry;
+      mc[i-SIKE_P434_R3_NWORDS_FIELD] = v;
+      v = u;
+      u = t;
+      t = 0;
+  }
+
+  /* `carry` isn't read after this, but it's still a necessary argument to the macro */
+    /* cppcheck-suppress unreadVariable */
```

I'm not sure if the awslc CI uses cppcheck. You might be able to delete these two lines. (`/* cppcheck-suppress unreadVariable */` looks like a comment, but it's actually an instruction to cppcheck, telling it to suppress the error it generates about the unread variable on the next line.)